test/common/cdp-driver.js

```diff
@@ -0,0 +1,238 @@
+            if (frameIdToContextId[frameId] == info.executionContextId) {
+                delete frameIdToContextId[frameId];
+                break;
+            }
+        }
+    });
+}
+
+
+// helper functions for testlib.py which are too unwieldy to be poked in from Python
+function getFrameExecId(frame) {
+    var frameId = frameNameToFrameId[frame];
+    if (!frameId)
+        throw Error(`Frame ${frame} is unknown`);
+    var execId = frameIdToContextId[frameId];
+    if (!execId)
+        throw Error(`Frame ${frame} (${frameId}) has no executionContextId`);
+    return execId;
+}
+
+function expectLoadFrame(name, timeout) {
+    return new Promise((resolve, reject) => {
+        let tm = setTimeout( () => reject("timed out waiting for frame load"), timeout );
+
+        // we can only have one Page.frameNavigated() handler, so let our handler above resolve this promise
+        frameWaitName = name;
+        new Promise((fwpResolve, fwpReject) => { frameWaitPromiseResolve = fwpResolve })
+            .then(() => {
+                // For the frame to be fully valid for queries, it also needs the corresponding
+                // executionContextCreated() signal. This might happen before or after frameNavigated(), so wait in case
+                // it happens afterwards.
+               function pollExecId() {
+                    if (frameIdToContextId[frameNameToFrameId[name]]) {
+                        clearTimeout(tm);
+                        resolve();
+                    } else {
+                        setTimeout(pollExecId, 100);
+                    }
+                }
+                pollExecId();
+            });
+    });
+}
+
+/**
+ * SSL handling
+ */
+
+// secure by default; tests can override to "continue"
+// https://chromedevtools.github.io/devtools-protocol/1-3/Security/#type-CertificateErrorAction
+var ssl_bad_certificate_action = "cancel";
+
+function setupSSLCertHandling(client) {
+    client.Security.enable();
+
+    client.Security.setOverrideCertificateErrors({override: true});
+    client.Security.certificateError(info => {
+        process.stderr.write(`CDP: Security.certificateError ${JSON.stringify(info)} (ignoring)\n`);
+        client.Security.handleCertificateError({ eventId: info.eventId, action: ssl_bad_certificate_action });
+    });
+}
+
+/**
+ * Main input/process loop
+ *
+ * Read one line with a JS expression, eval() it, and respond with the result:
+ *    success <JSON formatted return value>
+ *    fail <JSON formatted error>
+ * EOF shuts down the client.
+ */
+process.stdin.setEncoding('utf8');
+
+if (process.env["TEST_CDP_DEBUG"])
+    enable_debug = true;
+
+options = { };
+if (process.argv.length >= 3) {
+    options.port = parseInt(process.argv[2]);
+    if (!options.port) {
+        process.stderr.write("Usage: cdp-driver.js [port]\n");
+        process.exit(1);
+    }
+}
+
+CDP.New(options)
+    .then(target => {
+        target.port = options.port;
+        CDP(options)
```

Comment:
You can pass the `target` received from `CDP.New` into the `options` of `CDP()`. It probably won't make a difference because we'll only have one target most of the time.

Reply:
Fixed.